lib/std/compress/lzma2.zig

```diff
@@ -83,11 +83,17 @@ pub const AccumBuffer = struct {
             return error.CorruptInput;
         }
 
+        // ensure we have enough capacity for all new bytes.
+        // This prevents the buffer's memory from being reallocated (and freed)
+        // while we are still reading from it.
```

Comment:
This is a misleading comment because while this `ensureTotalCapacity` is reducing the number of allocations needed, it should not have any other effects.

Indeed, it works around #22906, but if that's what you're trying to note, then you should note that directly.